consumer/consumererror/signalerrors.go

```diff
@@ -24,20 +33,44 @@ func (err retryable[V]) Data() V {
 	return err.data
 }
 
+// Delay returns the time duration before the telemetry data should
+// be resent to the destination.
+func (err retryable[V]) Delay() time.Duration {
+	return err.delay
+}
+
+// RetryOption allows adding data to a retryable error,
+// e.g. the duration before sending should be retried.
+type RetryOption func(err *retryableCommon)
+
+func WithRetryDelay(delay time.Duration) RetryOption {
+	return func(err *retryableCommon) {
+		err.delay = delay
+	}
+}
+
 // Traces is an error that may carry associated Trace data for a subset of received data
 // that failed to be processed or sent.
 type Traces struct {
 	retryable[ptrace.Traces]
 }
 
 // NewTraces creates a Traces that can encapsulate received data that failed to be processed or sent.
-func NewTraces(err error, data ptrace.Traces) error {
-	return Traces{
+func NewTraces(err error, data ptrace.Traces, options ...RetryOption) error {
```

Comment:
This is a breaking change and should go through the regular deprecation process correct?

Reply:
I don't believe it is a breaking change since the new `options` parameter is variadic. You can see an example [here](https://github.com/evan-bradley/opentelemetry-collector/blob/issue-7047/consumer/consumererror/signalerrors_test.go#L28) where existing calls still work unmodified. That said, we would have to make a breaking change if we adopt approach 2 that I've [outlined here](https://github.com/open-telemetry/opentelemetry-collector/pull/9041#discussion_r1566386757).

Reply:
It's technically a breaking change: https://go.dev/play/p/ckLOeL-dtKM although one that we would rarely hit

Reply:
Since the actual impact is rare, I think doing this breaking once directly would be worth the 3-release-cycle deprecation process.

Reply:
@mx-psi That's a good catch. I agree with @TylerHelmuth that it's probably rare enough that it's not worth the extra churn involved with the deprecation process. Running a quick search through contrib, I suspect it will go unnoticed.

Reply:
Fine for me so long as we mark it as breaking on the changelog. We may want to update our breaking changes guidelines since these edge case-y breaking changes seem like a common situation

Reply:
I have a PR to add this specific case here: https://github.com/open-telemetry/opentelemetry-collector/pull/10041. If there are others changes we also want to call out, let me know and I can address them in follow-ups, or in that PR if they will affect this PR.

Reply:
Sorry, I think my message was a bit confusing. I still think we should consider this a breaking change, and I still think after 1.0 changes like these should be forbidden. What I meant is that I would be in favor of changing the guidelines on when to do a deprecation on pre-1.0 modules: I feel like we don't need to wait several releases to actually do a change like this while we are pre-1.0.

Reply:
I've added a breaking changelog entry that follows the guidelines established in #10041.